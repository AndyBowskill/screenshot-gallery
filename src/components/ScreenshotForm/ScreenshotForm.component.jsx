import React from 'react';
import './ScreenshotForm.style.css';

const ScreenshotForm = ({
  onInputWebsiteChange,
  onSaveButtonClick,
  isDisabled,
  buttonText,
}) => {
  return (
    <section>
      <div className='screenshotform flex'>
        <input
          type='text'
          placeholder='Please enter a website'
          name='inputWebsite'
          id='inputWebsite'
          onChange={onInputWebsiteChange}
        />
        <button disabled={isDisabled} onClick={onSaveButtonClick}>
          {buttonText}
        </button>
      </div>
    </section>
  );
};

export default ScreenshotForm;
