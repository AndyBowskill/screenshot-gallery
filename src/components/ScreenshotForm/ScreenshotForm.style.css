.screenshotform.flex {
  display: flex;
  justify-content: center;
  padding: 16px;
}

.screenshotform input {
  font-family: 'Didact Gothic', sans-serif;
  font-size: 1rem;
  padding: 16px;
  margin: 12px 4px 12px 12px;
  width: 100%;
  border: 1px solid #0074d9;
  border-radius: 5px;
}

.screenshotform button {
  font-family: 'Didact Gothic', sans-serif;
  font-size: 1rem;
  background-color: rgb(127, 219, 255);
  color: #0074d9;
  padding: 12px 24px;
  margin: 12px 12px 12px 4px;
  text-align: center;
  text-decoration: none;
  border: 1px solid #0074d9;
  cursor: pointer;
  border-radius: 5px;
  opacity: 0.9;
}

.screenshotform button:hover,
.screenshotform button[hover] {
  opacity: 1;
}

.screenshotform button:disabled,
.screenshotform button[disabled] {
  border: 1px solid #999999;
  background-color: #cccccc;
  color: #666666;
  cursor: default;
  opacity: 0.9;
}
