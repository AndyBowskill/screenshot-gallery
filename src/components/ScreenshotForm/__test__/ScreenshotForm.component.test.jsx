import React from 'react';
import renderer from 'react-test-renderer';
import ScreenshotForm from '../ScreenshotForm.component';

const onInputChange = (event) => {
  // Empty function
};

const onSaveButtonClick = () => {
  // Empty function
};

test('ScreenshotForm component displays correctly.', () => {
  const component = renderer.create(
    <ScreenshotForm
      onInputChange={onInputChange}
      onSaveButtonClick={onSaveButtonClick}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
