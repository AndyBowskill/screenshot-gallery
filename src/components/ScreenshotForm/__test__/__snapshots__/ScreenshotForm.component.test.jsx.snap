// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenshotForm component displays correctly. 1`] = `
<section>
  <div
    className="screenshotform flex"
  >
    <input
      id="inputWebsite"
      name="inputWebsite"
      onChange={[MockFunction]}
      placeholder="Please enter a website"
      type="text"
    />
    <button
      disabled={[MockFunction]}
      onClick={[MockFunction]}
    />
  </div>
</section>
`;
